python/ml4ir/base/data/tfrecord_writer.py

```diff
@@ -104,100 +118,94 @@ def write_from_df(
-    )
-    args = parser.parse_args(argv)
-
-    # Get all CSV files to be converted
+    # Get all CSV files to be converted, depending on user's arguments
     if args.csv_dir:
         csv_files: List[str] = glob.glob(os.path.join(args.csv_dir, "*.csv"))
     else:
-        csv_files: List[str] = [args.csv_file]
-
-    feature_config: FeatureConfig = parse_config(args.feature_config)
-
-    # Setup logging
-    logger: Logger = setup_logging()
+        csv_files: List[str] = args.csv_files
+    # Load feat config
+    feature_config: FeatureConfig = parse_config(MODES[args.tfmode], args.feature_config, logger)
 
     # Convert to TFRecord SequenceExample protobufs and save
-    file_count = 0
-    if args.convert_single_files:
+    if args.keep_single_files:
         # Convert each CSV file individually - better performance
         for csv_file in csv_files:
-            if args.tfrecord_dir:
-                tfrecord_file: str = os.path.join(
-                    args.tfrecord_dir, "file_{}.tfrecord".format(file_count)
-                )
-            else:
-                tfrecord_file: str = args.tfrecord_file
-
+            tfrecord_file: str = os.path.basename(csv_file).replace(".csv", '')
+            tfrecord_file: str = os.path.join(args.out_dir, "{}.tfrecord".format(tfrecord_file))
             write_from_files(
                 csv_files=[csv_file],
                 tfrecord_file=tfrecord_file,
                 feature_config=feature_config,
                 logger=logger,
+                tfrecord_type=MODES[args.tfmode]
             )
 
-            file_count += 1
     else:
         # Convert all CSV files at once - expensive groupby operation
-        if args.tfrecord_dir:
-            tfrecord_file: str = os.path.join(
-                args.tfrecord_dir, "file_{}.tfrecord".format(file_count)
-            )
-        else:
-            tfrecord_file: str = args.tfrecord_file
-
+        tfrecord_file: str = os.path.join(args.out_dir, "combined.tfrecord")
         write_from_files(
             csv_files=csv_files,
             tfrecord_file=tfrecord_file,
             feature_config=feature_config,
             logger=logger,
+            tfrecord_type=MODES[args.tfmode]
         )
 
 
+def define_arguments():
+    first_doc_line = __doc__.strip().split("\n")[0]
+    parser = ArgumentParser(description=first_doc_line)
+    parser.add_argument(
+        'tfmode',
+        choices=MODES,  # Choices with a dict, shows the keys
+        help="select between `sequence` and `example` to write tf.Example or tf.SequenceExample")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--csv-dir", type=str,  help="Directory with CSV files; every .csv file will be converted."
+    )
+    group.add_argument(
+        "--csv-files", type=str, nargs="+", help="A single or more (space separated) CSV files to be converted."
+    )
+    parser.add_argument(
```

Comment:
Here, this is another difference: to maintain the API somewhat simple, for writing the files the user can not choose a name. If several files, we will write in the out-dir changing the filenames extension from csv to tfrecords. I believe this is better from the previous solution because you did not know the association from fnames to numbered outputs (out1.tfrecords, out2.tfrecord,..). When though, a single combined tfrecords from multiple files is to be written, it is called `combined.tfrecord`.